src/Dofs/ConstraintHandler.jl

```diff
@@ -61,6 +61,78 @@ function __to_components(c)
+with the facet that minimizes
+```math
+U(\boldsymbol{q}(\boldsymbol{x})) = \int_{\Gamma^f}
+\left[\boldsymbol{q}(\boldsymbol{x}) \cdot \boldsymbol{n} - f(\boldsymbol{x}, t, \boldsymbol{n}) \right]^2 \mathrm{d}\Gamma
+```
+with ``\boldsymbol{q}(\boldsymbol{x}) = \boldsymbol{N}^f_j(\boldsymbol{x}) a_j^f``.
+Finding the stationary point by taking the directional derivative in all possible directions,
+``\delta\boldsymbol{q}(\boldsymbol{x})``, yields,
+```math
+0 = \lim_{\epsilon\rightarrow 0}\frac{\partial}{\partial \epsilon}
+U\big(\boldsymbol{q}(\boldsymbol{x}) + \epsilon \delta\boldsymbol{q}(\boldsymbol{x})\big)
+ = 2\int_{\Gamma^f} \left[\boldsymbol{q}(\boldsymbol{x}) \cdot \boldsymbol{n} - f(\boldsymbol{x}, t, \boldsymbol{n})\right]
+    \delta\boldsymbol{q}(\boldsymbol{x}) \cdot \boldsymbol{n}\ \mathrm{d}\Gamma
+```
+Inserting the FE-approximations,
+``\boldsymbol{q}(\boldsymbol{x}) = \boldsymbol{N}^f_j(\boldsymbol{x}) a_j^f`` and
+``\delta\boldsymbol{q}(\boldsymbol{x}) = \boldsymbol{N}^f_i(\boldsymbol{x}) c_i^f``,
+and using the arbitrariness of ``c_i^f``, yields the linear equation system,
+```math
+\underbrace{\int_{\Gamma^f} \left[\delta\boldsymbol{N}^f_i \cdot \boldsymbol{n}^f\right]\left[\boldsymbol{N}^f_j \cdot \boldsymbol{n}^f\right]\ \mathrm{d}\Gamma}_{K^f_{ij}}\ a_j^f
+= \underbrace{\int_{\Gamma^f} \left[\delta\boldsymbol{N}^f_i \cdot \boldsymbol{n}^f\right] f(\boldsymbol{x}, t, \boldsymbol{n})\ \mathrm{d}\Gamma}_{f^f_i}
+```
+determining the coefficients to be prescribed, ``a_j^f``.
+
+# H(curl) interpolations
+For H(curl), we want to prescribe the tangential flux, ``\boldsymbol{q}_\mathrm{t} = \boldsymbol{f}(\boldsymbol{x}, t, \boldsymbol{n})``.
+To that end, we want to find the degree of freedom values, ``a_j^f``, associated
+with the the facet that minimizes
+```math
+U(\boldsymbol{q}(\boldsymbol{x})) = \int_{\Gamma^f}
+\left\vert\left\vert
+    \boldsymbol{q}(\boldsymbol{x}) \times \boldsymbol{n} - \boldsymbol{f}(\boldsymbol{x}, t, \boldsymbol{n})
+\right\vert\right\vert^2
+\mathrm{d}\Gamma
+```
+with ``\boldsymbol{q}(\boldsymbol{x}) = \boldsymbol{N}^f_j(\boldsymbol{x}) a_j^f``.
+Similar to for ``H(\mathrm{div})``, we find the minimum by setting the directional derivative equal to zero,
+which after inserting the FE-approximations yields the linear equation system,
+```math
+\underbrace{\int_{\Gamma^f} \left[\delta\boldsymbol{N}^f_i \times \boldsymbol{n}^f\right]\cdot\left[\boldsymbol{N}^f_j \times \boldsymbol{n}^f\right]\ \mathrm{d}\Gamma}_{K^f_{ij}}\ a_j^f
+= \underbrace{\int_{\Gamma^f} \left[\delta\boldsymbol{N}^f_i \times \boldsymbol{n}^f\right]\cdot \boldsymbol{f}(\boldsymbol{x}, t, \boldsymbol{n})\ \mathrm{d}\Gamma}_{f^f_i}
+```
+determining the coefficients to be prescribed, ``a_j^f``
+"""
+mutable struct L2ProjectedDirichlet
+    const f::Function
+    const facets::OrderedSet{FacetIndex}
+    const field_name::Symbol
+    const qr_order::Int
+    # Created during `add!`
+    fv::Union{Nothing, FacetValues}
```

Comment:
Just to make sure: This differs from Dirichlet in the sense that it only works on subdomains with the same element type (i.e. 3D no mixed grids), right? Should we do this split automatically in `add!` as a followup and add an error for now?

I assume that this is not covered yet because the 3D elements are not merged yet?

Reply:
Will add support for that now that #1173 has been merged.

Reply:
I don't know if that arises from some derivations (I thought so initially, but doesn't seem like it based on what I found when writing the following: https://ferrite-fem.github.io/Ferrite.jl/previews/PR798/tutorials/maxwell_good_bad_ugly/#Theoretical-background)

Reply:
Had already added support for this :D, now it is also tested.